lib/puppet.rb

```diff
@@ -36,6 +38,27 @@ module Puppet
   require 'puppet/environments'
 
   class << self
+    # TODO: Handle the rubygems case?
+    local_locale_path = File.absolute_path('../locales', File.dirname(__FILE__))
+    posix_system_locale_path = File.absolute_path('../../../share/puppet/locale', File.dirname(__FILE__))
+    win32_system_locale_path = File.absolute_path('../../../../../puppet/share/puppet/locale', File.dirname(__FILE__))
+
+    if File.exist?(local_locale_path)
+      locale_path = local_locale_path
+    elsif File.exist?(win32_system_locale_path)
+      locale_path = win32_system_locale_path
+    elsif File.exist?(posix_system_locale_path)
+      locale_path = posix_system_locale_path
+    else
+      # We couldn't load our locale data. Possibly we're loaded as a rubygem or something?
+      locale_path = nil
+    end
+
+    if locale_path
+      GettextSetup.initialize(locale_path)
+      FastGettext.locale = GettextSetup.negotiate_locale(Locale.current.language)
```

Comment:
This is expected, and it does the same thing on all platforms. If we want the more specific info, there are other ways to interact with the API that are a bit more complicated but preserve the full name.